### PR TITLE
mime type checking and improve arweave/ipfs file naming

### PR DESCRIPTION
**Pull request by mjh1**
Fixes https://github.com/livepeer/catalyst-api/issues/231

### CODE REVIEW

File: pipeline/mist.go [hunk 1]
```diff
@@ -75,6 +84,19 @@ func (m *mist) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
...
 
+func isVideo(source string) (bool, error) {
+	header, err := clients.GetArweaveHeaders(source)
+	if err != nil {
+		return false, err
+	}
+	contentType := header.Get("Content-Type")
+	mediaType, _, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		return false, err
```
Comment (mjh1) [inline pipeline/mist.go:95]: @thomshutt do you know if we're ok to require the content-type header or should we fail open instead?

Comment (thomshutt) [reply pipeline/mist.go:99]: I definitely don't think we should rely on it being set correctly

Comment (mjh1) [reply pipeline/mist.go]: yeah make sense, reverted that bit 👍

File: clients/arweave_ipfs_s3.go [hunk 2]
```diff
@@ -51,3 +51,11 @@ func IsArweaveOrIPFSURL(arweaveOrIPFSURL string) bool {
 
 	return strings.Contains(u.Host, "arweave") || strings.Contains(u.Host, "w3s.link") || strings.Contains(u.Path, "ipfs")
 }
+
+func GetArweaveHeaders(arweaveURL string) (http.Header, error) {
```
Comment (thomshutt) [inline clients/arweave_ipfs_s3.go:55]: This isn't Arweave-specific, so can probably have a more generic name

File: pipeline/mist.go [hunk 3]
```diff
@@ -39,7 +40,15 @@ func (m *mist) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 	// Arweave URLs don't support HTTP Range requests and so Mist can't natively handle them for segmenting
 	// This workaround copies the file from Arweave to S3 and then tells Mist to use the S3 URL
 	if clients.IsArweaveOrIPFSURL(job.SourceFile) {
-		newSourceURL, err := inSameDirectory(job.TargetURL, "source", "arweave-source.mp4")
+		if !isVideo(job.RequestID, job.SourceFile) {
```
Comment (thomshutt) [inline pipeline/mist.go:43]: Does it make sense to do this check for all videos (i.e not just Arweave / IPFS ones)?

Comment (mjh1) [reply pipeline/mist.go]: sounds reasonable to me 👍

File: pipeline/mist.go [hunk 4]
```diff
@@ -36,10 +38,17 @@ func (m *mist) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 	job.SegmentingTargetURL = segmentingTargetURL.String()
 	log.AddContext(job.RequestID, "segmented_url", job.SegmentingTargetURL)
 
+	if !isVideo(job.RequestID, job.SourceFile) {
+		return nil, fmt.Errorf("source was not a video: %s", job.SourceFile)
```
Comment (mjh1) [inline pipeline/mist.go:42]: does anyone know offhand how this error would end up appearing to the user? do we have a special way to signal "bad request" type errors to the user?

Comment (thomshutt) [reply pipeline/mist.go:42]: Not at the moment, this will just show up in logs

**Pull request merged**